leopard.go

```diff
@@ -18,12 +15,37 @@ func init() {
 type leoRSFF8Codec struct{}
 
 func (l leoRSFF8Codec) Encode(data [][]byte) ([][]byte, error) {
-	return leopard.Encode(data)
+	return encode(data)
+}
+
+func encode(data [][]byte) ([][]byte, error) {
+	enc, err := reedsolomon.New(len(data), len(data))
```

Comment:
From https://github.com/celestiaorg/rsmt2d/pull/103#issuecomment-1243519416 it looks like we should be re-using the instance for same sizes. We need to be careful when using the instances in go-routines though.